libraries/acme.rb

```diff
@@ -31,13 +31,13 @@ def acme_client
 
   directory = new_resource.dir.nil? ? node['acme']['dir'] : new_resource.dir
 
-  contact = new_resource.contact.nil? ? node['acme']['contact'] : new_resource.contact
+  contact = (new_resource.contact.nil? || new_resource.contact.empty?) ? node['acme']['contact'] : new_resource.contact
 
   @client = Acme::Client.new(private_key: private_key, directory: directory)
 
   if node['acme']['private_key'].nil?
     acme_client.new_account(contact: contact, terms_of_service_agreed: true)
-    node.normal['acme']['private_key'] = private_key.to_pem
```

Comment:
I'm in doubt about this change from `normal` to `default` as I believe it will remove the persistency of the generated private key. Could you check that please?

Reply:
Ah, good point about persistence. Though, would writing the key to disk make more sense instead of keeping it in the node attributes? `node.normal` is strongly discouraged now.

Reply:
Yes that makes sense!